Delete With Predicate Panics

Steps to reproduce:

1. Set the featureFlag for deleteWithPredicate to true:
2. Navigate to the dashboard
3. `delete data` button in the header options
4. Select a bucket that has the tag key `cpu`
4. Set filter to `cpu = cpu1 AND cpu != cpu2`
5. Submit delete request

```
panic: runtime error: index out of range [1] with length 1

goroutine 4525 [running]:
github.com/influxdata/influxdb/tsdb/tsm1.(*predicateState).Set(...)
	/Users/asalem/go/src/github.com/influxdb/tsdb/tsm1/predicate.go:325
github.com/influxdata/influxdb/tsdb/tsm1.(*predicateMatcher).Matches(0xc001bca260, 0x16104ce5, 0x1e, 0x1eb8df, 0x3500b00)
	/Users/asalem/go/src/github.com/influxdb/tsdb/tsm1/predicate.go:123 +0x2ca
github.com/influxdata/influxdb/tsdb/tsm1.(*Engine).DeletePrefixRange.func3(0x352bc80, 0xc0005ea800, 0x0, 0x0)
	/Users/asalem/go/src/github.com/influxdb/tsdb/tsm1/engine_delete_prefix.go:145 +0x546
github.com/influxdata/influxdb/tsdb/tsm1.(*FileStore).Apply.func1(0xc001a71680, 0xc0009f3aa0, 0xc0006da100, 0x352bc80, 0xc0005ea800)
	/Users/asalem/go/src/github.com/influxdb/tsdb/tsm1/file_store.go:535 +0x102
created by github.com/influxdata/influxdb/tsdb/tsm1.(*FileStore).Apply
	/Users/asalem/go/src/github.com/influxdb/tsdb/tsm1/file_store.go:529 +0x114
exit status 2```

__Expected behavior:__
Delete data with predicate

__Actual behavior:__
Panic

--- a/tsdb/tsm1/predicate.go
+++ b/tsdb/tsm1/predicate.go
@@ -75,7 +75,15 @@ func NewProtobufPredicate(pred *datatypes.Predicate) (Predicate, error) {
 		if node.GetNodeType() == datatypes.NodeTypeTagRef {
 			switch value := node.GetValue().(type) {
 			case *datatypes.Node_TagRefValue:
-				locs[value.TagRefValue] = len(locs)
+				// Only add to the matcher locations the first time we encounter
+				// the tag key reference. This prevents problems with redundant
+				// predicates like:
+				//
+				//   foo = a AND foo = b
+				//   foo = c AND foo = d
+				if _, ok := locs[value.TagRefValue]; !ok {
+					locs[value.TagRefValue] = len(locs)
+				}
 			}
 		}
 	})

--- a/tsdb/tsm1/predicate_test.go
+++ b/tsdb/tsm1/predicate_test.go
@@ -86,6 +86,16 @@ func TestPredicate_Matches(t *testing.T) {
 		},
 
 		{
+			Name: "Logical And Matching Reduce (Simplify)",
+			Predicate: predicate(
+				andNode(
+					comparisonNode(datatypes.ComparisonEqual, tagNode("foo"), stringNode("bar")),
+					comparisonNode(datatypes.ComparisonNotEqual, tagNode("foo"), stringNode("bif")))),
+			Key:     "bucketorg,baz=bif,foo=bar,tag3=val3",
+			Matches: true,
+		},
+
+		{
 			Name: "Regex Matching",
 			Predicate: predicate(
 				comparisonNode(datatypes.ComparisonRegex, tagNode("tag3"), regexNode("...3"))),

